src/Redeemer.sol

```diff
@@ -247,23 +247,27 @@ contract Redeemer {
     }
 
     // @notice: Handles all necessary ETH conversion when redeeming a LST rather than direct ETH
+    // @param pool: The address of the pool to swap to
     // @param lst: The address of the token to swap to
     // @param amount: The amount of underlying to spend
     // @param swapMinimum: The minimum amount of lst to receive
     // @returns lent: The amount of underlying to be lent
     // @returns slippageRatio: The slippageRatio of the swap (1e18 based % to adjust swivel orders if necessary)
-    function convert(address lst, uint256 amount, uint256 swapMinimum) external authorized(admin) returns (uint256 lent, uint256 slippageRatio) {
-        // TODO: convert ETHWrapper to stateless proxy calls
+    function convert(address pool, address lst, uint256 amount, uint256 swapMinimum) external authorized(admin) returns (uint256 returned, uint256 slippageRatio) {
         address ETHWrapper = ILender(lender).ETHWrapper();
-        Safe.transferFrom(IERC20(lst), ETHWrapper, address(this), amount);
-        if (lst != address(0)) {
-            (lent, slippageRatio)  = IETHWrapper(ETHWrapper).swap(lst, ILender(lender).WETH(), amount, swapMinimum);
-            return (lent, slippageRatio);
-        }
-        // If the `lst` address is blank, no swap is necessary
-        else {
-            revert Exception(0, 0, 0, address(0), address(0)); // TODO: add error code
+
+        // Conduct the lend operation to acquire principal tokens
+        (bool success, bytes memory returndata) = ETHWrapper.delegatecall(
+            abi.encodeWithSignature('swap(address,address,address,uint256,uint256)', pool, lst, "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE", amount, swapMinimum));
```

Comment:
The same question here, why `"0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE"` and not `ILender(lender).WETH()`?